libs/builder/helpers/angular/get-component-inputs.ts

```diff
@@ -0,0 +1,55 @@
+import {ClassDeclaration, PropertyDeclaration} from 'ts-morph';
+
+import {forAllClasses} from '../typescript/class/for-all-classes';
+
+/**
+ * Retrieve all existing `@Input` of an Angular component, up through ascendant classes
+ *
+ * @param cls - ClassDeclaration
+ * @param condition - (cls: ClassDeclaration) => boolean
+ */
+export function getComponentInputs(
+	cls: ClassDeclaration,
+	condition: (cls: ClassDeclaration) => boolean = classHasNoParent,
+): PropertyDeclaration[] {
+	const allComponentAndParentsInputs: PropertyDeclaration[][] = [];
+
+	forAllClasses(cls, (c: ClassDeclaration) => {
+		const isBaseClass = c !== cls && condition(c);
```

Comment:
From my understanding of the `forAllClasses` function, this condition is needed to know when to stop `while` loop once the base class is reached.
I could have just inlined the `classHasNoParent()` helper function instead of setting a parameter maybe. I can update is you want to :)

Reply:
Yes, that's correct. I added the ability to return `true` in order to stop checking parent classes. This is used, for example, to extract class member documentation from parent classes. But in this case, I think it's unnecessary because we still need to check all classes and collect their inputs anyway.

Reply:
Yep, I removed that code